### PR TITLE
Fixes #3428 Add snapshot save/load for OverwriteParameterSet

### LINKED ISSUE #3428: OverwriteParameterSet serialization (project save/load)
## Parent PRD

#3422

## What to build

Implement serialization for OverwriteParameterSet to enable project save/load. This follows the existing snapshot mapper patterns used in PKSim.

**Snapshot classes:**
- Create snapshot class for OverwriteParameterSet containing Name, IsDefault, ExtendedProperties, and ParameterValues
- OverwriteParameterSet will be part of the Compound snapshot

**Snapshot mappers:**
- Create `OverwriteParameterSetMapper` following existing mapper patterns
- Integrate with existing Compound snapshot serialization
- Handle the ParameterValues collection serialization

**Simulation serialization:**
- Integrate with existing simulation snapshot

## Acceptance criteria

- [ ] Snapshot class for OverwriteParameterSet exists
- [ ] OverwriteParameterSetMapper implemented
- [ ] Compound snapshot includes OverwriteParameterSets collection
- [ ] Simulation snapshot includes OverwriteParameterSetSelections
- [ ] Round-trip tests pass (save → load → verify equality)
- [ ] Existing compounds without OverwriteParameterSets load correctly (empty collection)

## Blocked by

- #3427

## User stories addressed

- User story 17: undo/redo support (persistence foundation)
- User story 20: uncommitted parameter changes persisted when saving

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #3428 Add snapshot save/load for OverwriteParameterSet

## Summary
- Project serialization had been handled in a prior PR
- Add snapshot classes (`OverwriteParameterSet`, `ParameterValue`, `OverwriteParameterSetSelection`) and dedicated mappers for serializing `OverwriteParameterSet` on `Compound` and `OverwriteParameterSetSelections` on `Simulation`
- Each domain↔snapshot conversion uses a dedicated mapper (`OverwriteParameterSetMapper`, `ParameterValueMapper`, `OverwriteParameterSetSelectionMapper`) following the established codebase pattern, reusing `ExtendedPropertyMapper` from OSPSuite.Core
- Update `CompoundMapper` and `SimulationMapper` to persist and restore the new structures during snapshot save/load

## Test plan
- [x] All 5119 existing tests pass with 0 failures
- [x] Round-trip tests for `OverwriteParameterSetMapper` (model → snapshot → model)
- [x] Round-trip test for `OverwriteParameterSetSelectionMapper` including compound/parameter set resolution from project
- [x] Edge case tests: empty collections, missing compound, missing parameter set
- [x] `CompoundMapper` test verifies overwrite parameter sets appear in snapshot
- [x] `SimulationMapper` tests verify selections are saved/loaded correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added support for overwrite parameter sets on compounds, allowing custom parameter value definitions.
  * Added the ability to associate overwrite parameter sets with simulations.
  * Enhanced snapshot functionality to preserve and restore parameter set configurations and their relationships.

### CODE REVIEW

File: src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   public OverwriteParameterSetSelectionMapper(IOSPSuiteLogger logger)
+   {
+      _logger = logger;
+   }
+
+   public override async Task<SnapshotOverwriteParameterSetSelection> MapToSnapshot(ModelOverwriteParameterSetSelection selection, PKSimProject project)
+   {
+      return await SnapshotFrom(selection, snapshot =>
+      {
+         snapshot.CompoundName = selection.CompoundName;
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs:27]: Why do we need the compound name at all in the parameter set?

Comment (rwmcintosh) [reply src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs:27]: I don't follow. 

This is the selection done in the simulation. So I think for each compound, you have 0 or 1 overwrite sets. They match up by names

Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs:27]: Sorry I thought this was the object in the compound

File: src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      var compound = project.BuildingBlockByName<ModelCompound>(snapshot.CompoundName);
+      if (compound == null)
+      {
+         _logger.AddWarning(PKSimConstants.Error.SimulationTemplateBuildingBlockNotFoundInProject(snapshot.CompoundName, nameof(ModelCompound)));
+         return Task.FromResult<ModelOverwriteParameterSetSelection>(null);
+      }
+
+      var overwriteParameterSet = compound.OverwriteParameterSets.FirstOrDefault(x => x.IsNamed(snapshot.OverwriteParameterSetName));
+      if (overwriteParameterSet == null)
+         return Task.FromResult<ModelOverwriteParameterSetSelection>(null);
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs:46]: that we should probabluy log also

File: src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using ModelParameterValue = OSPSuite.Core.Domain.Builder.ParameterValue;
+using SnapshotParameterValue = PKSim.Core.Snapshots.ParameterValue;
+
+namespace PKSim.Core.Snapshots.Mappers;
+
+public class ParameterValueMapper : SnapshotMapperBase<ModelParameterValue, SnapshotParameterValue>
+{
+   public override async Task<SnapshotParameterValue> MapToSnapshot(ModelParameterValue parameterValue)
+   {
+      // We will only use path and value for this application
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: Comments like this are stupid. I can see that we only use path and value. Comment the WHY not the HOW

File: src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      return await SnapshotFrom(parameterValue, snapshot =>
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: return await is alwyas suspicisous for me. Are we doing this often in teh code base?

File: src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      {
+         snapshot.Path = parameterValue.Path.ToString();
+         snapshot.Value = parameterValue.Value.GetValueOrDefault();
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: I think this is wrong. We will have a value if it's not set. Why are we using GetValueOrDefault

**Review by msevestre — CHANGES_REQUESTED** [review]

File: src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs (hunk 4 above)
Comment (rwmcintosh) [reply src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: quite a few places in the snapshot mappers

File: src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs (hunk 5 above)
Comment (rwmcintosh) [reply src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: Yeah, we should count on there being a value if it's in the OverwriteParameterSet. That's what it does.

[the remark this answers is not in the document]
Comment (rwmcintosh) [reply src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: Updated because we are expecting the value to always be present. It's by definition for a ParameterValue in OvewriteParameterSet.

File: src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rwmcintosh) [reply src/PKSim.Core/Snapshots/Mappers/OverwriteParameterSetSelectionMapper.cs:46]: added a log

File: src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs (hunk 4 above)
Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: well let' remove the new one at least. return is fine (unless you are doing a try catch)

Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: only 9 lol. that's ok not that bad

Comment (rwmcintosh) [reply src/PKSim.Core/Snapshots/Mappers/ParameterValueMapper.cs]: Ok, I think this is what you meant.
return await => return
method is no longer `async`

File: src/PKSim.Core/Snapshots/OverwriteParameterSetSelection.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+namespace PKSim.Core.Snapshots
+{
+   public class OverwriteParameterSetSelection
+   {
+      public string CompoundName { get; set; }
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/OverwriteParameterSetSelection.cs:8]: should not we marked this with the [Requuired] attribute? 
All of theses fields are actually required.

File: src/PKSim.Core/Snapshots/ParameterValue.cs [hunk 7]
```diff
@@ -0,0 +1,7 @@
+namespace PKSim.Core.Snapshots;
+
+public class ParameterValue
+{
+   public string Path { get; set; }
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/ParameterValue.cs:8]: same with Path and Value. This should be [Required]

**Review by msevestre — CHANGES_REQUESTED** [review]
as I was looking for sthg else in code, I realzie that we are not using the required stuff. It prevents the json from being serailzied or deserialized if wrong. That way we neer have to worry about an edited file
